core/src/main/java/de/danoeh/antennapod/core/sync/SynchronizationQueue.java

```diff
@@ -0,0 +1,96 @@
+package de.danoeh.antennapod.core.sync;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.util.ArrayList;
+
+import de.danoeh.antennapod.core.ClientConfig;
+import de.danoeh.antennapod.net.sync.model.EpisodeAction;
+
+public class SynchronizationQueue {
+
+    public static final String NAME = "synchronization";
+
+    private static SharedPreferences getSharedPreferences() {
+        return ClientConfig.applicationCallbacks.getApplicationInstance()
+                .getSharedPreferences(NAME, Context.MODE_PRIVATE);
```

Comment:
Hmm, if you make this a singleton with something like `getInstance(context)` or by passing the context to the methods, this would no longer depend on the ugly `ClientConfig` class. I would love to get rid of `ClientConfig` one day. Also makes it easier to move the class to a "sync" module (in the future).

Reply:
The class now creates SharedPreferences in constructor. There is nothing static anymore.
The only call that was used from a static context `clearQueue` forced me to change some other methods. please see for yourself if this is a good thing.